forc-plugins/forc-client/src/cmd/call.rs

```diff
@@ -229,18 +254,63 @@ pub struct Command {
     /// The external contract addresses to use for the call
     /// If none are provided, the call will automatically populate external contracts by making a dry-run calls
     /// to the node, and extract the contract addresses based on the revert reason
-    #[clap(long, alias = "contracts")]
+    #[clap(long, alias = "contracts", help_heading = "CONTRACT")]
     pub external_contracts: Option<Vec<ContractId>>,
 
-    /// The output format to use; possible values: default, raw
-    #[clap(long, default_value = "default")]
+    /// Output format for the call result
+    #[clap(long, default_value = "default", help_heading = "OUTPUT")]
     pub output: OutputFormat,
 
-    /// Output call receipts
-    #[clap(long, short = 'r', alias = "receipts")]
+    /// Show transaction receipts in the output
+    #[clap(long, short = 'r', alias = "receipts", help_heading = "OUTPUT")]
     pub show_receipts: bool,
 }
 
+impl Command {
+    /// Validate the command and determine the CLI operation
+    pub fn validate_and_get_operation(&self) -> Result<Operation, String> {
+        // Case 1: List functions
+        if self.list_functions {
+            if self.abi.is_none() {
+                return Err("ABI is required when using --list-functions".to_string());
+            }
+            return Ok(Operation::ListFunctions {
+                contract_id: (*self.address).into(),
+                abi: self.abi.clone().expect("ABI should be validated"),
+            });
+        }
+
+        // Case 2: Direct transfer with amount
+        if self.function.is_none() && self.call_parameters.amount > 0 {
+            if !matches!(self.mode, ExecutionMode::Live) {
+                return Err("Direct transfers are only supported in live mode".to_string());
+            }
```

Comment:
simpler to read if we are only comparing against one variant 

```suggestion
           if self.mode != ExecutionMode::Live {
               return Err("Direct transfers are only supported in live mode".to_string());
           }
```

Reply:
https://github.com/FuelLabs/sway/pull/7056/commits/0e48d88b2d574cdc0fe7147ff2c15cc8c3c19f1b